### PR TITLE
Add support for JWT authentication

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Closes https://github.com/crate/crate/issues/14238

Supersedes https://github.com/crate/crate/pull/15640 (each commit here has changes related to comments in the old PR)

First commit is combination of https://github.com/crate/crate/pull/15515 and part of https://github.com/crate/crate/pull/15562 (another part is exposure of props in sys.users table)

Other commits pretty much 1:1 to already reviewed PR-s.

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
> ```
> Error:  Failures: 
> Error:    RoleManagementIntegrationTest.testAlterNonExistingUserThrowsException:206 [Expected an exception with errorCode 40410 but NullPointerException has a different code] 
> expected: 40410
>  but was: 5000
> ```
> 
> will take another look

pushed a fixup.

File: docs/sql/statements/alter-role.rst [hunk 1]
```diff
@@ -55,6 +55,17 @@ are supported to alter an existing user account:
     For security reasons it is recommended to authenticate as ``crate`` using a
     client certificate.
 
+:jwt:
+  JWT properties map (``iss`` and ``username``) entered as a string literal.
+  e.g.::
+
+     ALTER USER john WITH (jwt = {"iss" = 'new_issuer', "username" = 'new_username'})
```
Comment (mfussenegger) [inline docs/sql/statements/alter-role.rst:62]:
```suggestion
     ALTER USER john WITH (jwt = {"iss" = 'new_issuer', "username" = 'john.smith'})
```

To better illustrate that it's the same user but the name can diverge

File: docs/sql/statements/create-user.rst [hunk 2]
```diff
@@ -79,3 +79,26 @@ The following ``user_parameter`` are supported to define a new user account:
...
      CREATE USER john password 'foo'
+
+.. vale off
+
+.. _create-user-jwt:
+
+:jwt:
+  JWT properties map ('iss' and 'username') entered as string literal. e.g.::
+
+     CREATE USER john WITH (jwt = {"iss" = 'https://login.company.org/keys', "username" = 'test@company.org'})
```
Comment (mfussenegger) [inline docs/sql/statements/create-user.rst:90]: Please use example.com as domain name, see https://www.rfc-editor.org/rfc/rfc6761.html

**Review by mfussenegger — APPROVED** [review]

File: docs/sql/statements/create-user.rst (hunk 2 above)
Comment (BaurzhanSakhariev) [reply docs/sql/statements/create-user.rst]: updated and squashed into first commit

**Pull request merged**